dotcom-rendering/src/components/AppsEpic.importable.tsx

```diff
@@ -0,0 +1,125 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import type { IGetEpics__Result } from '@guardian/bridget/Acquisitions';
+import {
+	body as bodySizes,
+	from,
+	headline,
+	palette,
+	remSpace,
+	textSans,
+} from '@guardian/source-foundations';
+import { useEffect, useState } from 'react';
+import { getAcquisitionsClient, getUserClient } from '../lib/bridgetApi';
+import { EpicContent } from './EpicContent.apps';
+
+export const AppsEpic = () => {
+	const [isPremium, setIsPremium] = useState<boolean | undefined>(undefined);
+	const [maybeEpic, setMaybeEpic] =
+		useState<IGetEpics__Result['success']>(undefined);
+
+	useEffect(() => {
+		void getUserClient()
+			.isPremium()
+			.then(setIsPremium)
+			.catch(() => undefined);
+	}, []);
+
+	useEffect(() => {
+		void getAcquisitionsClient()
+			.getEpics()
+			.then((me) => {
+				setMaybeEpic(me);
+			})
+			.catch(() => undefined);
+	}, []);
+
+	if (navigator.onLine) {
```

Comment:
Because this is an external, mutable piece of state, I think we may want to avoid reading from it directly in rendering code[^1]. If we do I think the component will be out of sync with this state: for example, the user may be online when this is rendered, then go offline, but the component will still be rendered as though they were online.

 I think the recommendation is to use `useSyncExternalStore` for this[^2][^3]. That said, this will cause the whole epic to appear and disappear when the user is online or offline, so we may want to do something more subtle like disable the buttons while they're offline?

[^1]: https://react.dev/learn/keeping-components-pure
[^2]: https://react.dev/learn/you-might-not-need-an-effect#subscribing-to-an-external-store
[^3]: https://react.dev/reference/react/useSyncExternalStore#subscribing-to-a-browser-api

Reply:
Ok, I think that sounds reasonable, but we may want to discuss the disappearing behaviour with UX as a follow-up? Looking at the code is there a scenario at the moment in AR where this would disappear once it's first been rendered?